anime_downloader/sites/animesvostfr.py

```diff
@@ -0,0 +1,247 @@
+            episodeurlapi2 = ''
+            
+            soup = helpers.soupify(helpers.get(episodeurl, cf=True))
+            #logger.debug( "results: {}".format(soup) )
+            
+            iframe = soup.select_one("iframe")
+            if iframe:
+                episodeurl2 = iframe['src']
+                
+                logger.debug( "_get_sources episodeurl2:%s", episodeurl2 )
+                
+                episodeurlapi2 = episodeurl2.replace('/v/', '/api/source/' )
+                
+            
+            logger.debug( "_get_sources episodeurlapi1:%s", episodeurlapi1 )    
+            logger.debug( "_get_sources episodeurlapi2:%s", episodeurlapi2 )   
+            
+            #data = helpers.post(episodeurlapi, params={'r' : '', 'd' : 'youtubedownloader.cx' }, referer=episodeurl, cf=True).json()
+            #data = helpers.post(episodeurlapi, params={'r' : '', 'd' : '' }, referer=episodeurl).json()
+            #data = helpers.post(episodeurlapi, params={'r' : '' }, referer=episodeurl).json()
+            
+            data = None
+            if episodeurlapi2:
+                data = helpers.post(episodeurlapi2, referer=episodeurl2).json()
+            elif episodeurlapi1:
+                data = helpers.post(episodeurlapi1, referer=episodeurl1).json()
+                
+            if data:
+                #logger.debug( "data: {}".format(data) )
+                logger.debug( 'Success: %s', str(data['success']) )
+                logger.debug( "data: {}".format(data['data']) )
+                
+                url_quality = {}
+                
+                for d in data['data']:
+                    #logger.debug( "data: {}".format(d) )
+                    
+                    quality = d['label']
+                    url = d['file']
+                    
+                    if not quality in url_quality:
+                        #self.__emails.append(email)
+                        url_quality[quality] = url
+                 
+                  
+                logger.debug( "quality: {}".format(url_quality) )
+                logger.debug( "quality by def: {}".format(self.quality) ) 
+                logger.debug( "quality fallback: {}".format(self._parent._fallback_qualities) )
+                
+                quality_chosen = None
+                if self.quality in url_quality:
+                    quality_chosen = url_quality[self.quality]
+                else:
+                    for quality in self._parent._fallback_qualities:
+                        if quality in url_quality:
+                            quality_chosen = url_quality[quality_chosen]
+                            
+                if not quality_chosen:
+                    quality_chosen = url_quality[0]   
+                    
+                logger.debug( "quality_chosen: %s", str(quality_chosen) );
+                    
+                if not quality_chosen:
+                    raise NotFoundError      
```

Comment:
Anime-downloader has a built in function when no links are found, use that instead. To do it simply do `return ''`

Reply:
ok i have to do it